server/src/main/java/org/elasticsearch/repositories/RepositoriesMetrics.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.repositories;
+
+import org.elasticsearch.telemetry.metric.LongCounter;
+import org.elasticsearch.telemetry.metric.LongHistogram;
+import org.elasticsearch.telemetry.metric.MeterRegistry;
+
+public class RepositoriesMetrics {
+
+    public static RepositoriesMetrics NOOP = new RepositoriesMetrics(MeterRegistry.NOOP);
+
+    private static final String METRIC_REQUESTS_COUNT = "es.repositories.requests.count";
+    private static final String METRIC_EXCEPTIONS_COUNT = "es.repositories.exceptions.count";
+    private static final String METRIC_THROTTLES_COUNT = "es.repositories.throttles.count";
+    private static final String METRIC_OPERATIONS_COUNT = "es.repositories.operations.count";
+    private static final String METRIC_UNSUCCESSFUL_OPERATIONS_COUNT = "es.repositories.operations.unsuccessful.count";
+    private static final String METRIC_EXCEPTIONS_HISTOGRAM = "es.repositories.exceptions.histogram";
+    private static final String METRIC_THROTTLES_HISTOGRAM = "es.repositories.throttles.histogram";
+
+    private final LongCounter requestCounter;
+
+    private final LongCounter exceptionCounter;
+    private final LongCounter throttleCounter;
+    private final LongCounter operationCounter;
+    private final LongCounter unsuccessfulOperationCounter;
+    private final LongHistogram exceptionHistogram;
+    private final LongHistogram throttleHistogram;
+
+    public RepositoriesMetrics(MeterRegistry meterRegistry) {
```

Comment:
This is the new class. Everything else is cascading changes

Reply:
Considering its structure is there a value in converting this class to a record and replace constructor with a static factory?
This way we will not need to declare getters

Reply:
Sure I changed it to be a record.